pandas/tests/io/formats/test_format.py

```diff
@@ -1269,18 +1269,60 @@ def test_to_string_specified_header(self):
             df.to_string(header=['X'])
 
     def test_to_string_no_index(self):
-        df = DataFrame({'x': [1, 2, 3], 'y': [4, 5, 6]})
 
-        df_s = df.to_string(index=False)
-        expected = "x  y\n1  4\n2  5\n3  6"
+        dfs = [
 
-        assert df_s == expected
+            # ints
+            DataFrame({'x': [1, 2, 3], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, 33], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [4, 5, 6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [4, 5, -6]}),
+            DataFrame({'x': [11, 22, -33], 'y': [44, 55, -66]}),
+
+            # floats
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [4, 5, 6]}),
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [0.4, 0.5, 0.6]}),
+            DataFrame({'x': [0.1, 0.2, -0.3], 'y': [0.4, 0.5, -0.6]}),
+        ]
+
+        exs = [
+
+            # ints
+            " x  y\n 1  4\n 2  5\n 3  6",
+            "  x  y\n 11  4\n 22  5\n 33  6",
+            "  x  y\n 11  4\n 22  5\n-33  6",
+            "  x  y\n 11  4\n 22  5\n-33 -6",
+            "  x   y\n 11  44\n 22  55\n-33 -66",
+
+            # floats
+            "   x  y\n 0.1  4\n 0.2  5\n-0.3  6",
+            "   x    y\n 0.1  0.4\n 0.2  0.5\n-0.3  0.6",
+            "   x    y\n 0.1  0.4\n 0.2  0.5\n-0.3 -0.6",
+        ]
+
+        for df, expected in zip(dfs, exs):
+            df_s = df.to_string(index=False)
+            assert df_s == expected
```

Comment:
Definitely should use `pytest.mark.parametrize` for this.

Reply:
I updated the code style to match other tests in the same file.